java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/ArrowFlightJdbcDriver.java

```diff
@@ -247,8 +248,15 @@ private Map<Object, Object> getUrlsArgs(String url)
     final String extraParams = uri.getRawQuery(); // optional params
 
     final Map<String, String> keyValuePairs = UrlParser.parse(extraParams, "&");
-    resultMap.putAll(keyValuePairs);
+    final Map<String, String> lowerCaseKeyValuePairs = lowerCaseKeys(keyValuePairs);
+    resultMap.putAll(lowerCaseKeyValuePairs);
 
     return resultMap;
   }
+
+  private Map<String, String> lowerCaseKeys(final Map<String, String> keyValues) {
+    Map<String, String> resultMap = new HashMap<>();
+    keyValues.forEach((k, v) -> resultMap.put(k.toLowerCase(), v));
```

Comment:
This converts connection string properties to lower case but what about if someone uses the Properties object and has keys in the wrong case?

Reply:
done